Samples/resource-owner-password-credential/Angular2/src/newoidc/Controllers/AccountController.cs

```diff
@@ -0,0 +1,512 @@
+        }
+
+        //
+        // POST: /Account/Login
+        [HttpPost]
+        [AllowAnonymous]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null)
+        {
+            ViewData["ReturnUrl"] = returnUrl;
+            if (ModelState.IsValid)
+            {
+                // This doesn't count login failures towards account lockout
+                // To enable password failures to trigger account lockout, set lockoutOnFailure: true
+                var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, lockoutOnFailure: false);
+                if (result.Succeeded)
+                {
+                    _logger.LogInformation(1, "User logged in.");
+                    return RedirectToLocal(returnUrl);
+                }
+                if (result.RequiresTwoFactor)
+                {
+                    return RedirectToAction(nameof(SendCode), new { ReturnUrl = returnUrl, RememberMe = model.RememberMe });
+                }
+                if (result.IsLockedOut)
+                {
+                    _logger.LogWarning(2, "User account locked out.");
+                    return View("Lockout");
+                }
+                else
+                {
+                    ModelState.AddModelError(string.Empty, "Invalid login attempt.");
+                    return View(model);
+                }
+            }
+
+            // If we got this far, something failed, redisplay form
+            return View(model);
+        }
+
+        //
+        // GET: /Account/Register
+        [HttpGet]
+        [AllowAnonymous]
+        public IActionResult Register(string returnUrl = null)
+        {
+            ViewData["ReturnUrl"] = returnUrl;
+            return View();
+        }
+
+        // custom logout
+        [HttpGet]
+        [Route("api/account/logout")]
+        public async Task<string> LogOut()
+        {
+            await _signInManager.SignOutAsync();
+            _logger.LogInformation(4, "User logged out.");
+            return "{done:success}";
+        }
+        // custom register
+        [Route("api/account/register")]
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IdentityResult> RegisterHttp([FromBody]RegisterViewModel dto)
+        {
+            var user = new ApplicationUser { UserName = dto.Email, Email = dto.Email };
+            var result = await _userManager.CreateAsync(user, dto.Password);
+            return result;
+        }
+
+        //custom external access use anything solid rather than 123456
+        [Route("api/account/externalAccess")]
+        [AllowAnonymous]
+        [HttpGet]
+        public IActionResult ExtLogin(string provider, string returnUrl = "http://localhost:58056/connect/authorize?client_id=myClient&redirect_uri=http://localhost:58056/signin-oidc&response_type=id_token&scope=openid&nonce=123456")
```

Comment:
Omagad, that's terrible. The authorization request URL must be created from the JS side and implement things correctly (with a real random nonce and a state to prevent XSRF attacks)